clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -313,6 +317,13 @@ private KafkaProducer(ProducerConfig config, Serializer<K> keySerializer, Serial
                 config.ignore(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG);
                 this.valueSerializer = valueSerializer;
             }
+
+            // load interceptors and make sure they get clientId
+            userProvidedConfigs.put(ProducerConfig.CLIENT_ID_CONFIG, clientId);
+            List<ProducerInterceptor<K, V>> interceptorList = (List) (new ProducerConfig(userProvidedConfigs)).getConfiguredInstances(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG,
```

Comment:
Hmm, what happens if the user configures an interceptor with the wrong key and value type parameters? It won't be caught here and we'll probably get a ClassCastException when we invoke one of the methods. It would be good to decide what behaviour we want, document it and add a test (unless this is all done already).